iroh-one/src/core.rs

```diff
@@ -0,0 +1,155 @@
+use crate::{config::Config, mem_p2p, mem_store};
+use anyhow::{anyhow, Result};
+use iroh_api::Api;
+use iroh_gateway::{bad_bits::BadBits, metrics};
+use iroh_metrics::MetricsHandle;
+use iroh_rpc_client::Client as RpcClient;
+use std::sync::Arc;
+
+#[cfg(feature = "uds-gateway")]
+use iroh_one::uds;
+use iroh_resolver::racing::RacingLoader;
+use iroh_rpc_types::Addr;
+#[cfg(feature = "uds-gateway")]
+use tempdir::TempDir;
+use tokio::sync::RwLock;
+use tokio::task;
+
+pub struct Core<'a> {
```

Comment:
`Core` is a really unfortunate and non-descriptive name, just `Iroh` or `One` (given this lives inside of `iroh-one` atm) would make more sense to me